src/dom-manipulation.ts

```diff
@@ -19,109 +20,112 @@ export function appendFlightInformationToFlightInfoContainer(flight) {
   }
 }
 
-function createNewFlightInfoRow(flight, flightInfoDiv) {
-  if (flight[0]) {
+// double null checks to be removed in observable brance
+
+function createNewFlightInfoRow(flight: IFlights, flightInfoDiv: HTMLElement) {
+  if (flight.icao24) {
     flightInfoDiv.innerHTML += `
   <div id="${
-    flight[0]
+    flight.icao24
   }" class="single-flight grid grid-cols-4 py-1 text-[0.8rem] md:grid-cols-5 lg:grid-cols-6 xl:grid-cols-7">
-      <span>${flight[1] || "None"}</span>
+      <span>${flight.callsign || "None"}</span>
       <span class="hidden xl:contents">${convertMeterPerSecondToKilomentersPerHour(
-        flight[11] ?? 0.0
+        flight.vertical_rate ?? 0.0
       )}km/h</span>
       <span>${convertMeterPerSecondToKilomentersPerHour(
-        flight[9] ?? 0
+        flight.velocity ?? 0
       )}km/h</span>
       <span class="hidden md:contents">${calculateDirection(
-        flight[10] ?? 0
+        flight.true_track ?? 0
       )}</span>
-      <span>${flight[10] ?? 0}°</span>
-      <span class="hidden lg:contents">${flight[7] ?? 0.0}m</span>
+      <span>${flight.true_track ?? 0}°</span>
+      <span class="hidden lg:contents">${flight.baro_altitude ?? 0.0}m</span>
       <button id="${
-        flight[0] + (flight[1] ?? 0)
+        flight.icao24 + (flight.callsign ?? 0)
       }" class="track-button uppercase font-bold cursor-pointer border-none">track</button> 
     </div>`;
   }
 }
 
-function appendExistingFlightInfoRow(flight) {
-  const exsitingFlightInfoRow = document.getElementById(flight[0]);
+function appendExistingFlightInfoRow(flight: IFlights) {
```

Comment:
strongly type your functions that don't return a value e.g. void.

Reply:
I have added :void to string type my non returning fuctions